src/test/run-pass/issue-34053.rs

```diff
@@ -17,7 +17,14 @@ impl Drop for A {
 }
```

Comment:
I feel like we need a test that demonstrates the runtime semantics of constants with DROP. In particular, the destructor does execute when you (e.g.) access `BAR` or `A::BAZ`, right?

I want some test that shows when the destructor runs and tests that it did indeed execute (i.e., by observing some effect it had on a global counter).

Reply:
Ah, right, this was the odd thing about the RFC amendment.